lib/Backend/BackendUtils.cpp

```diff
@@ -433,10 +433,47 @@ void allocatePlaceholders(const ContiguousPlaceholders &placeholders,
 void allocateActivations(const glow::IRFunction::InstListTy &instrs,
                          MemoryAllocator &allocator,
                          glow::runtime::SymbolTableTy &symbolTable) {
+
+  // Gather allocation/deallocation sequence.
```

Comment:
@mciprian13 

While debugging the issue we found on our side, I found it useful to add a command-line option telling the memory allocation logic to never reuse memory for activations (so that you can easily inspect the value of each activation and to make sure that two activations never use the same memory).

Achieving this is pretty easy: Simply insert all the deallocations at the very end of the allocList. This effectively makes all activations alive for the whole duration of the IRFunction and thus they never reuse memory.

If it is not too much trouble, I'd suggest you add such a command-line debug option and the corresponding logic here. Should be just about 10-15 lines of code I think.

Reply:
@opti-mix So you mean to restore the old functionality of the `reuseActivationsMemory`? Just to confirm what you want when this flag is false is to have non-overlapping activations both within same function and also between different functions, right? As you said that is easily doable.

Reply:
Yes, I mean exactly this. Thanks for implementing it.